Sming/SmingCore/Network/TcpClient.cpp

```diff
@@ -53,6 +53,14 @@ TcpClient::~TcpClient()
 	}
 }
 
+bool TcpClient::connect(const char * server, int port, boolean useSsl /* = false */, uint32_t sslOptions /* = 0 */)
+{
+	if (isProcessing()) return false;
+
+	state = eTCS_Connecting;
+	return TcpConnection::connect(server, port, useSsl, sslOptions);
```

Comment:
This might actually call the present function again making stack ovf by recursion. Compiler will probably choose const char* version and not string version. Also even if you make it call the string version you are doing the isprocessing checks again... i just see no added benefit here maybe if it was the other way aroung e.g. the string fct would call the const char* fct. Do you agree? (Sorry for typing badly i'm on mobile phone)